### ISSUE TITLE
Cache doesn't work with get_or_create even if object exists

### ISSUE THREAD

**Issue by vatsalj** [Suor/django-cacheops#236]
Hi,

I have a `UserTemplate` model having a OneToOne relation with `User` and a simple `CharField`. When querying using `UserTemplate.objects.get(user=user)`, I can see the object is getting served from cache, but when using `UserTemplate.objects.get_or_create(user=user)`, cache is not getting hit. Fresh object is retrieved from the database and inserted into the cache with the same key. I identified the issue with django debug toolbar and verified with redis-cli monitor. Is there something else that needs to be done with `get_or_create` to make it work with cacheops?

**Comment by Suor** [comment]
A queryset in "for_write" mode is not cached. This is done to prevent caching the internals of `.get_or_create()`, `.update_or_create()`, `.bulk_create()`, `.delete()` and `Model.save()`, as well as always sending SELECT FOR UPDATE queries to db. Strictly speaking only SELECT FOR UPDATE requires it, but considering CAVEATS this is not entirely safe to make write queries based on cached data, also many of those specific cached queries most probably won't be hit - you usually insert a new object only once and delete it once too. 

Another issue is slave lag: cache could be populated from lagging slave, which might prevent correct behavior or even write back stale data. `.get_or_create()` won't recreate recently deleted obj and `.update_or_create()` can write updated old object.

However, `.get_or_create()` requires such protection the least of all of the above. And many CAVEATS were lifted since this rule was implemented. So I will think of special casing it, though this idea still looks a bit unsafe.

For the time being you can reimplement `.get_or_create()` on specific queryset subclass or just write a function helper, where you set `self._for_write = True` after first get.

**Comment by Suor** [comment]
Possible both backwards compatible and safe solution is to add special `'get_or_create'` op besides `'get'`.

**Comment by Suor** [comment]
Going to use current behavior as default. It did get huge traction and a workaround is accessible anyway.
